pkg/objstore/objtesting/foreach.go

```diff
@@ -67,4 +68,21 @@ func ForeachStore(t *testing.T, testFn func(t testing.TB, bkt objstore.Bucket))
 	} else {
 		t.Log("THANOS_SKIP_S3_AWS_TESTS envvar present. Skipping test against S3 AWS.")
 	}
+
+	// Optional Azure
+	if _, ok := os.LookupEnv("THANOS_SKIP_AZURE_TESTS"); !ok {
```

Comment:
Let's add this to CI for now here https://github.com/improbable-eng/thanos/blob/master/.circleci/config.yml#L36, until we don't have CI configured for azure (will do that later)